### PR TITLE
Fix contributed profile icons

### LINKED ISSUE #182303: Contributed icon in JS debug terminal stopped working
See https://github.com/microsoft/vscode-js-debug/issues/975#issuecomment-1544573154

I can reproduce:

image

### PR DESCRIPTION
**Pull request by tisilent**
Fix  #182303 .

The little bug has returned.
企业微信截图_16842321008527

### CODE REVIEW

**Review by Tyriar — APPROVED** [review]
Thanks @tisilent! I made the fix a little more generic so it works with all extensions, thanks for the investigation

**Review by amunger — APPROVED** [review]

**Comment by tisilent** [comment]
@Tyriar You better  👍 ,I learned.

**Comment by tisilent** [comment]
Wait..There are still some issues.

**Comment by Tyriar** [comment]
@tisilent oh? It was working when I tested it out?

File: src/vs/workbench/contrib/terminal/browser/terminalInstance.ts [hunk 1]
```diff
@@ -485,8 +485,9 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 				const defaultProfile = (await this._terminalProfileResolverService.getDefaultProfile({ remoteAuthority: this.remoteAuthority, os }));
 				this.shellLaunchConfig.executable = defaultProfile.path;
 				this.shellLaunchConfig.args = defaultProfile.args;
-				this.shellLaunchConfig.icon = defaultProfile.icon;
-				this.shellLaunchConfig.color = defaultProfile.color;
+				// Only use default icon and color if they are undefined in the SLC
+				this.shellLaunchConfig.icon ??= defaultProfile.icon;
+				this.shellLaunchConfig.color ??= defaultProfile.color;
```
Comment (tisilent) [inline src/vs/workbench/contrib/terminal/browser/terminalInstance.ts:490]: @Tyriar  The 'New Terminal' action requires default values.

Comment (tisilent) [reply src/vs/workbench/contrib/terminal/browser/terminalInstance.ts:490]: Issues:
企业微信截图_1684722768662
'New Terminal' using terminal icon, pwsh icon should be used.

After modify, the test is working:
`
// Only use default icon and color if they are undefined in the SLC
if (!this.shellLaunchConfig.isExtensionOwnedTerminal) {
	this.shellLaunchConfig.icon = defaultProfile.icon;
	this.shellLaunchConfig.color = defaultProfile.color;
} else {
	this.shellLaunchConfig.icon ??= defaultProfile.icon;
	this.shellLaunchConfig.color ??= defaultProfile.color;
}
`

**Pull request merged**